cmd/oras/root/push.go

```diff
@@ -41,6 +44,7 @@ import (
 type pushOptions struct {
 	option.Common
 	option.Packer
+	option.Platform
```

Comment:
Should add this to `Packer` options

Reply:
But if this is added to `Packer`, we need to change `oras attach` since it already has a `--platform` flag.

Reply:
1. Both `attach` and `push` should support customized platform. 
2. It's inconsistent if we use `--platform` for customizing config platform in `push`, while `--platform` for selecting subject platform in `attach`

Reply:
Let not merge it to `Packer` since this PR is all about `push`